pkgs/applications/misc/somebar/default.nix

```diff
@@ -0,0 +1,43 @@
+{ lib
+, stdenv
+, writeText
+, fetchurl
+, meson
+, ninja
+, pkg-config
+, wayland
+, pango
+, wayland-protocols
+, conf ? null
+, patches ? [ ]
+}:
+
+let
+  # There is a configuration in src/config.def.hpp, which we use by default
+  configFile = if lib.isDerivation conf || builtins.isPath conf then conf else "src/config.def.hpp";
+in
+
+stdenv.mkDerivation rec {
+  pname = "somebar";
+  version = "1.0.0";
+
+  src = fetchurl {
+    url = "https://git.sr.ht/~raphi/somebar/archive/${version}.tar.gz";
+    sha256 = "sha256-lCBjnwUeJ4R6jVy36KjFkUG7pn6WYA+Xkb0l43Fc0Mw=";
+  };
+
+  nativeBuildInputs = [ meson ninja pango pkg-config wayland wayland-protocols ];
+
+  # Allow for custom patches
+  inherit patches;
```

Comment:
Is there a reason to do it explicitly for this package?  Overrides should work for patches without it. See here for an example https://ryantm.github.io/nixpkgs/using/overrides/

Reply:
> Is there a reason to do it explicitly for this package? Overrides should work for patches without it. See here for an example https://ryantm.github.io/nixpkgs/using/overrides/

I was inspired by the [dwm](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/applications/window-managers/dwm/default.nix#L34) package, which does it this way. The thought is that instead of using normal overrides, you instead use overrideAttrs. I already implemented the `conf` variable to be able to be overriden by overrideAttrs, so the user can override the config file path which often is needed.

Reply:
Actually, I think I just realized I might have mixed overrideAttrs with overrides, and thus I think you are right that this isn't needed. I have removed this section in the latest commit (with the current version there are basically no patches avaliable anyway so this isn't needed for now anyway).